src/components/modals/data-table-modal.js

```diff
@@ -30,183 +32,179 @@ import {Clock} from 'components/common/icons/index';
-    // https://github.com/micho/jQuery.preventMacBackScroll
-    // This prevents scroll when reaching the topmost or leftmost
-    // positions of a container.
+const HeaderRenderer = React.memo(({columnIndex, data, rowIndex, style}) => {
+  const item = data[columnIndex];
 
-    // react-data-grid canvas element can be scrolled
-    const canvas = this._root.querySelector('.react-grid-Canvas');
+  return (
+    <div className="header" style={{...style, ...tagContainerStyle}}>
+      <div style={{display: 'flex', alignItems: 'center'}}>
+        <div>
+          {item.type === 'timestamp' ? <Clock height="16px" /> : null}
+        </div>
+        {item.name}
+      </div>
+      <div>
+        <FieldToken type={item.type} />
+      </div>
+    </div>
+  );
+});
+
+const CellRenderer = React.memo(({columnIndex, data, rowIndex, style}) => {
+  const item = data[rowIndex][columnIndex];
+
+  return (
+    <div
+      className="cell"
+      style={style}
+      key={`${rowIndex}-${columnIndex}`}>
+      {item}
+    </div>
+  );
+});
 
-    // If canvas can not be scrolled left anymore when we try to scroll left
-    const prevent_left = e.deltaX < 0 && canvas.scrollLeft <= 0;
-    // If canvas can not be scrolled up when we try to scroll up
-    const prevent_up = e.deltaY < 0 && canvas.scrollTop <= 0;
+export class DataTableModal extends Component {
 
-    if (prevent_left || prevent_up) {
-      e.preventDefault();
-    }
-  };
+  headerGrid = React.createRef();
+
+  datasetsSelector = props => props.datasets;
+  dataIdSelector = props => props.dataId;
+  activeDatasetSelector = createSelector(
+    this.datasetsSelector,
+    this.dataIdSelector,
+    (datasets, dataId) => datasets[dataId]
+  );
+  columnsSelector = createSelector(
+    this.activeDatasetSelector,
+    dataset => dataset.fields.filter(({name}) => name !== '_geojson')
+  );
+  rowsSelector = createSelector(
+    this.activeDatasetSelector,
+    dataset => dataset.data
+  );
+
+  headerRowHeight = () => 72;
+  cellRowHeight = () => 48;
+  columnWidth = index => COLUMN_SIZE[this.columnsSelector(this.props)[index].type] || DEFAULT_COLUMN_WIDTH;
+  onRowScrolling = ({scrollLeft}) => this.headerGrid.current.scrollTo({scrollLeft});
 
   render() {
-    const {datasets, dataId, showDatasetTable} = this.props;
+    const {showDatasetTable} = this.props;
+
+    const datasets = this.datasetsSelector(this.props);
+    const dataId = this.dataIdSelector(this.props);
 
     if (!datasets || !dataId) {
       return null;
     }
 
-    const activeDataset = datasets[dataId];
-    // TODO: this should be all data
-    const rows = activeDataset.data;
-    const columns = activeDataset.fields
-      .map((field, i) => ({
-        ...field,
-        key: i,
-        headerRenderer: <FieldHeader {...field} />,
-        resizable: true,
-        formatter:
-          field.type === ALL_FIELD_TYPES.boolean ? BooleanFormatter : undefined
```

Comment:
The `formatter` here is to make sure Boolean value renders correctly, without a specific formatter, boolean value can't be displayed

[on master]
<img width="627" alt="Screen Shot 2019-06-21 at 12 26 32 AM" src="https://user-images.githubusercontent.com/3605556/59905564-6d187e80-93bb-11e9-89df-8f74ad976415.png">

[in this pr]
<img width="417" alt="Screen Shot 2019-06-21 at 12 25 24 AM" src="https://user-images.githubusercontent.com/3605556/59905553-638f1680-93bb-11e9-8587-cc384dd4dfa1.png">